sdks/java/testing/load-tests/src/main/java/org/apache/beam/sdk/loadtests/LoadTest.java

```diff
@@ -81,36 +87,46 @@
    * Runs the load test, collects and publishes test results to various data store and/or console.
    */
   public PipelineResult run() throws IOException {
-    long testStartTime = System.currentTimeMillis();
+    Timestamp timestamp = Timestamp.now();
 
     loadTest();
 
     PipelineResult result = pipeline.run();
     result.waitUntilFinish();
 
-    LoadTestResult testResult = LoadTestResult.create(result, metricsNamespace, testStartTime);
+    String testId = UUID.randomUUID().toString();
+    List metrics = readMetrics(timestamp, result, testId);
 
-    ConsoleResultPublisher.publish(testResult);
+    ConsoleResultPublisher.publish(metrics, testId, timestamp.toString());
 
     if (options.getPublishToBigQuery()) {
-      publishResultToBigQuery(testResult);
+      publishResultsToBigQuery(metrics);
     }
     return result;
   }
 
-  private void publishResultToBigQuery(LoadTestResult testResult) {
+  private List<NamedTestResult> readMetrics(Timestamp timestamp, PipelineResult result, String testId) {
+    MetricsReader reader = new MetricsReader(result, metricsNamespace);
+
+    NamedTestResult runtime = NamedTestResult
+        .create(testId, timestamp.toString(), "runtime_sec",
+            (reader.getEndTimeMetric("runtime") - reader.getStartTimeMetric("runtime")) / 1000D);
+
+    NamedTestResult totalBytes = NamedTestResult
+        .create(testId, timestamp.toString(), "total_bytes_count",
+            reader.getCounterMetric("totalBytes.count"));
+
+    return Arrays.asList(runtime, totalBytes);
+  }
+
+  private void publishResultsToBigQuery(List<TestResult> testResults) {
```

Comment:
Type should be `List<NamedTestResult>`, since you're using the schema from that class.